jbpm-designer-backend/src/main/java/org/jbpm/designer/repository/vfs/RepositoryDescriptorProvider.java

```diff
@@ -81,8 +84,13 @@ public RepositoryDescriptor getRepositoryDescriptor(String repositoryAlias,
 
     private RepositoryDescriptor buildAndRegister(Repository repository,
                                                   String branchName) {
-        String repoUri = repository.getRoot().toURI().replaceFirst("://.*?@",
-                                                                   "://" + branchName + "@");
+
+        if (!repository.getDefaultBranch().isPresent()) {
```

Comment:
Not strictly necessary, but this would be a good place to use `Optional.orElseThrow`.